src/AlgebraicGeometry/Schemes/AffineAlgebraicSet/Objects/Methods.jl

```diff
@@ -66,3 +66,88 @@ function Base.show(io::IO, X::AffineAlgebraicSet)
   end
 end
 
+########################################################
+# (2) give back k-points (if finitely many)
+########################################################
+@doc raw"""
+    rational_point_coordinates(I::MPolyIdeal)
+
+Over the base field $k$, return the $k$-coordinates of the point 
+corresponding to a maximal ideal $I \in k[x_1,\dots,x_n]$, which 
+describes a $k$-point. 
+If $I$ is not maximal or does not describe a point with coordinates in the 
+field $k$, an error exception results.
+
+# Examples
+```jldoctest
+julia> R, (x, y) = QQ[:x, :y]
+(Multivariate polynomial ring in 2 variables over QQ, QQMPolyRingElem[x, y])
+
+julia> I = ideal(R, [x-1,y-3])
+Ideal generated by
+  x - 1
+  y - 3
+
+julia> rational_point_coordinates(I)
+2-element Vector{QQFieldElem}:
+ 1
+ 3
+
+```
+"""
+function rational_point_coordinates(I::MPolyIdeal)
+  R=base_ring(I)
+  o = degrevlex(gens(R))
+  LG = leading_ideal(I;ordering=o)
+  dim(LG)==0 || error("Ideal does not describe finite set of points")
```

Comment:
better to use `@req` here, and also two lines below

Reply:
done